server/datastore/mysql/migrations/tables/20230202224725_CreateHostDiskEncryptionKeysTable.go

```diff
@@ -0,0 +1,33 @@
+package tables
+
+import (
+	"database/sql"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230202224725, Down_20230202224725)
+}
+
+func Up_20230202224725(tx *sql.Tx) error {
+	// `decryptable` can be NULL to signal that we have fetched the key but
+	// we don't know yet if we can decrypt it or not, the related index is
+	// to aid querying for this scenario by taking adventage of MySQL's IS
+	// NULL optimization:
+	// https://dev.mysql.com/doc/refman/5.7/en/is-null-optimization.html
+	_, err := tx.Exec(`
+	  CREATE TABLE IF NOT EXISTS host_disk_encryption_keys (
```

Comment:
This table `host_disk_encryption_keys` must be added to the list of host-related tables, so that it gets properly cleared when the host is deleted (that's in `server/datastore/mysql/hosts.go`, look for `hostRefs` package-level variable). You will probably need to add a test case in `hosts_test.go` for `DeleteHosts`, as it checks that all host refs tables are first populated, then cleared after a delete.

Reply:
what a great catch, thank you! Fixed.